packages/apify/src/crawlers/browser_crawler.ts

```diff
@@ -9,30 +11,40 @@ import {
     handleRequestTimeout,
 } from './crawler_utils';
 import { BasicCrawler, BasicCrawlerOptions, CrawlingContext, HandleFailedRequest } from './basic_crawler';
-import { ProxyConfiguration } from '../proxy_configuration';
-import { RequestList } from '../request_list';
-import { RequestQueue } from '../storages/request_queue';
-import { Request } from '../request';
+import { ProxyConfiguration, ProxyInfo } from '../proxy_configuration';
 import { BrowserLaunchContext } from '../browser_launchers/browser_launcher';
 import { Session } from '../session_pool/session';
 import { Awaitable, Dictionary } from '../typedefs';
 
-export interface BrowserCrawlingContext<Page = unknown, Response = unknown> extends CrawlingContext<Response> {
-    browserController: BrowserController;
+export interface BrowserCrawlingContext<
+    Page extends CommonPage = CommonPage,
+    Response = unknown,
+    ProvidedController = BrowserController
+> extends CrawlingContext<Response> {
+    browserController: ProvidedController;
     page: Page;
 }
 
 export type BrowserHook<
     Context = BrowserCrawlingContext,
-    GoToOptions extends Record<PropertyKey, any> = Dictionary
+    GoToOptions extends Record<PropertyKey, any> | undefined = Dictionary
 > = (crawlingContext: Context, gotoOptions: GoToOptions) => Awaitable<void>;
-export type BrowserHandlePageFunction<Context = BrowserCrawlingContext> = (context: Context) => Awaitable<void>;
+
+export type BrowserHandlePageFunction<Context extends BrowserCrawlingContext = BrowserCrawlingContext> = (context: Context) => Awaitable<void>;
+
 export type GotoFunction<
     Context = BrowserCrawlingContext,
-    GoToOptions extends Record<PropertyKey, any> = Dictionary
-> = (context: Context, gotoOptions: GoToOptions) => Awaitable<any>;
-
-export interface BrowserCrawlerOptions extends Omit<BasicCrawlerOptions, 'handleRequestFunction'> {
+    GoToOptions extends Record<PropertyKey, any> | undefined = Dictionary
+> = (context: Context, gotoOptions?: GoToOptions) => Awaitable<any>;
+
+export interface BrowserCrawlerOptions<
```

Comment:
The type for BrowserPool is..a headache for sure, but it's actually much easier to use than it seems. All you have to do is pass in the `browserPlugin` in an object to the first parameter, and the rest is inferred! This is also why I made every generic argument in this interface optional, to hopefully make its usage less of a headache. Although, depending how we proceed with splitting up crawlers, this might end up completely different anyways, right?